flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/FileIOChannel.java

```diff
@@ -154,20 +155,25 @@ private static String randomString(Random random) {
 	public static final class Enumerator {
 
 		private final File[] paths;
-		
+
 		private final String namePrefix;
 
-		private int counter;
+		private int localCounter;
+
+		private static AtomicInteger globalCounter = new AtomicInteger();
```

Comment:
Minor (optional) comment: In the (undocumented) code style that Flink follows implicitly, the static member fields are the first in the class, before the instance member fields.